File: features/discovery/home/src/main/java/com/tokopedia/home/beranda/helper/DateHelper.java
package com.tokopedia.home.beranda.helper;

import java.text.SimpleDateFormat;
import java.util.Date;

/**
 * Created by errysuprayogi on 3/23/18.
 */

public class DateHelper {

    public static Date getExpiredTime(String expiredTimeString) {
        SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZZZZ");
        try {
            return format.parse(expiredTimeString);
        } catch (Exception e) {
            e.printStackTrace();
            return new Date();
        }
    }

    public static boolean isExpired(long serverTimeOffset, Date expiredTime) {
        Date serverTime = new Date(System.currentTimeMillis());
        serverTime.setTime(
                serverTime.getTime() + serverTimeOffset
        );
        return serverTime.after(expiredTime);
    }
}

File: features/content/play_common/src/main/java/com/tokopedia/play_common/widget/playBannerCarousel/helper/DateHelper.kt
package com.tokopedia.play_common.widget.playBannerCarousel.helper

import java.text.SimpleDateFormat
import java.util.*

object DateHelper {
    fun getExpiredTime(expiredTimeString: String?): Date {
        val format = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZZZZ")
        return try {
            format.parse(expiredTimeString)
        } catch (e: Exception) {
            e.printStackTrace()
            Date()
        }
    }

    fun isExpired(serverTimeOffset: Long, expiredTime: Date?): Boolean {
        val serverTime = Date(System.currentTimeMillis())
        serverTime.time = serverTime.time + serverTimeOffset
        return serverTime.after(expiredTime)
    }
}
